### ISSUE TITLE
Add performance validations to language server extended services

### ISSUE THREAD

**Issue by IMS94** [ballerina-platform/ballerina-lang#33151]
**Description:**
With more and more extended language server services getting introduced to the language server, number of requests served by LS at a given time increases. As of now, when a user makes a change to a document, apart from usual completion/codeaction requests, several other requests come from the vscode side. Examples are the diagram overview, executor positions and package components requests can be considered.

Since some of these requests rely on compilation, vscode may become unresponsive (latency in diagnostic publishing, etc) when the source code becomes larger. In LS, we have introduced the cancellation support to address this. Since no similar mechanism is available for extended services yet, irrelevant requests may take up CPU while the source code has moved on from that state.

To identify this kind of behavior, a mechanism is required to identify performance deviations.

**Describe your task(s)**
Suggestion is to implement a special type of performance tests with the following steps.
1. Simulate opening a text document and sending a completion request at the given cursor position. Measure time taken for response.
2. Simulate the above steps while sending extended service requests at the same time. Measure time taken.

Perform the above steps several times to eliminate outliers and get the average response times. Then, compare for the time difference in those two operations. Check if the extended service has affected the core LS features (completions in this case). We can see if the response time has increased than a given percentage.

In summary, we take the LS only scenario as the baseline and compare the same while being interfered by extended services. We should propose a test framework (an abstract class may be) which can be extended by extended services and add their own tests.

**Comment by anupama-pathirage** [comment]
We are closing this issue due to inactivity. If you need further assistance or have additional input, feel free to reopen it. Thank you for helping us maintain a relevant and focused issue list!
